server/controllers/finance/fiscal.js

```diff
@@ -457,3 +461,198 @@ function notNullBalance(array) {
+        'Ecriture de solde des profits pour la cloture',
+        item.id,                        // account_id
+        debit, credit,                  // debit and credit
+        debit, credit,                  // debit_equiv and credit_equiv in enterprise currency
+        currencyId,                     // enterprise currency because data came from period total
+        userId                          // user id
+      ];
+
+      transaction.addQuery(sqlInsertJournal, profitParams);
+    });
+
+    // sold the charge exploitation
+    exploitation.charge.forEach(item => {
+
+      // charge has debitor sold
+      let value = item.debit - item.credit;
+
+      // inverted values for solding
+      let debit = value >= 0 ? 0 : Math.abs(value);
+      let credit = value >= 0 ? Math.abs(value) : 0;
+
+      let chargeParams = [
+        projectId,                      // project_id
+        fiscal.id,                      // fiscal_year_id
+        fiscal.number_of_months + 1,    // period_id
+        fiscal.end_date,                // date : the last date of the fiscal year
+        'Ecriture de solde des charges pour la cloture',
+        item.id,                        // account_id
+        debit, credit,                  // debit and credit
+        debit, credit,                  // debit_equiv and credit_equiv in enterprise currency
+        currencyId,                     // enterprise currency because data came from period total
+        userId                          // user id
+      ];
+
+      transaction.addQuery(sqlInsertJournal, chargeParams);
+    });
+
+    // be sure to have accounts to sold, either profit or charge
+    if (exploitation.profit.length || exploitation.charge.length) {
+      // the resultat: profit - charge
+      let value = resultat.global;
+
+      // debit if benefits or credit if loss
+      let debit = value >= 0 ? 0 : Math.abs(value);
+      let credit = value >= 0 ? Math.abs(value) : 0;
+
+      let resultParams = [
+        projectId,                      // project_id
+        fiscal.id,                      // fiscal_year_id
+        fiscal.number_of_months + 1,    // period_id
+        fiscal.end_date,                // date : the last date of the fiscal year
+        'Ecriture du resultat lors de la cloture',
+        accountId,
+        debit, credit,                  // debit and credit
+        debit, credit,                  // debit_equiv and credit_equiv in enterprise currency
+        currencyId,                     // enterprise currency because data came from period total
+        userId                          // user id
+      ];
+
+      transaction.addQuery(sqlInsertJournal, resultParams);
+    }
+
+    return transaction.execute();
+  })
+  .then(() => {
+    // update the fiscal year
+    const sql = `UPDATE fiscal_year SET locked = 1 WHERE id = ?;`;
+    return db.exec(sql, [id]);
+  })
+  .then(rows => {
+    if (!rows.changedRows) {
+      const errorDescription = {
```

Comment:
In the future, I would prefer if you moved the error definition out of the promise chain.  Like this:
```js
// at the top of your module
const ERROR_FAILURE_CLOSING = {
  status : '',
  code : '',
  description : ''
};

// much later ...

promise
  .then(rows => {
     if (!rows.changedRows) {
        throw ERROR_FAILURE_CLOSING;
      }
   
    // etc
  });
```
It makes the code easier to read.